api/internal/core/entity/format_test.go

```diff
@@ -56,6 +56,39 @@ func TestNodesFormat(t *testing.T) {
 	assert.Contains(t, jsonStr, `"priority":10`)
 }
 
+func TestNodesFormat_ipv6(t *testing.T) {
+	// route data saved in ETCD
+	routeStr := `{
+		"uris": ["/*"],
+		"upstream": {
+			"type": "roundrobin",
+			"nodes": [{
+				"host": "::1",
+				"port": 80,
+				"weight": 0,
+				"priority":10
+			}]
+		}
+	}`
+
+	// bind struct
+	var route Route
+	err := json.Unmarshal([]byte(routeStr), &route)
+	assert.Nil(t, err)
+
+	// nodes format
+	nodes := NodesFormat(route.Upstream.Nodes)
+
+	// json encode for client
+	res, err := json.Marshal(nodes)
+	assert.Nil(t, err)
+	jsonStr := string(res)
+	assert.Contains(t, jsonStr, `"weight":0`)
+	assert.Contains(t, jsonStr, `"port":80`)
+	assert.Contains(t, jsonStr, `"host":"::1"`)
```

Comment:
according to the code, the host should be `[::1]`?

Reply:
https://apisix.apache.org/docs/apisix/admin-api/#request-body-parameters-3
In the 'nodes' field, it is mentioned that IPv6 addresses should be enclosed in square brackets.

Reply:
No brackets are needed when using object mode.